tests/envs/test_pybullet_blocks.py

```diff
@@ -0,0 +1,498 @@
+"""Test cases for PyBulletBlocksEnv."""
+
+import numpy as np
+import pytest
+from predicators.src.envs import PyBulletBlocksEnv
+from predicators.src.structs import Object, State, Action
+from predicators.src.ground_truth_nsrts import _get_predicates_by_names
+from predicators.src import utils
+from predicators.src.settings import CFG
+from predicators.tests.conftest import longrun
+
+_GUI_ON = False  # toggle for debugging
+_EXPOSED_PYBULLET_ENV = None  # only create once, since init is expensive
+
+
+class _ExposedPyBulletBlocksEnv(PyBulletBlocksEnv):
+
+    @property
+    def block_type(self):
+        """Expose the block type."""
+        return self._block_type
+
+    @property
+    def robot(self):
+        """Expose the robot, which is a static object."""
+        return self._robot
+
+    @property
+    def Pick(self):
+        """Expose the Pick parameterized option."""
+        return self._Pick
+
+    @property
+    def Stack(self):
+        """Expose the Stack parameterized option."""
+        return self._Stack
+
+    @property
+    def PutOnTable(self):
+        """Expose the PutOnTable parameterized option."""
+        return self._PutOnTable
+
+    def set_state(self, state):
+        """Forcibly reset the state."""
+        self._current_state = state
+        self._current_task = None
+        self._reset_state(state)
+
+    def get_state(self):
+        """Expose get state."""
+        return self._get_state()
+
+    def execute_option(self, option):
+        """Helper for executing a single option, updating the env state."""
+        # Note that since we want to use self._current_state, it's convenient
+        # to make this an environment method, rather than a helper function.
+        assert option.initiable(self._current_state)
+        # Execute the pick option.
+        while True:
+            if option.terminal(self._current_state):
+                break
+            action = option.policy(self._current_state)
+            self.step(action)
+        return self._current_state.copy()
+
+
+def _get_exposed_pybullet_env():
+    global _EXPOSED_PYBULLET_ENV  # pylint:disable=global-statement
+    if _EXPOSED_PYBULLET_ENV is None:
+        _EXPOSED_PYBULLET_ENV = _ExposedPyBulletBlocksEnv()
+    return _EXPOSED_PYBULLET_ENV
+
```

Comment:
There probably is a way to do this with pytest fixtures scoped to the module. So the fixture function is only invoked once for this entire test module.

https://docs.pytest.org/en/6.2.x/fixture.html

Doesn't matter too much though

Reply:
it would be cool to get this to work, which would allow us to remove the ugly `global` statement too!

Reply:
I can work on refactoring this when I work on my pybullet stuff, as I'll imagine needing to do something similar